src/ImageManipulation.php

```diff
@@ -1057,4 +1069,204 @@ protected function castDimension($value, $dimension)
+
+        return $file;
+    }
+
+    /**
+     * Initialise the attirbutes on this object. Should only be called imiditely after intialising the object.
+     * @param array $attributes
+     * @internal
+     */
+    public function initAttributes(array $attributes)
+    {
+        // This would normally be in a constructor, but there's no way for a trait to interact with that.
+        $this->attributes = $attributes;
+    }
+
+    /**
+     * Retrieve the value of an HTML attribute
+     * @param $name
+     * @return mixed|null
+     */
+    public function getAttribute($name)
+    {
+        $attributes = $this->getAttributes();
+
+        if (isset($attributes[$name])) {
+            return $attributes[$name];
+        }
+
+        return null;
+    }
+
+    /**
+     * Allows customization through an 'updateAttributes' hook on the base class.
+     * Existing attributes are passed in as the first argument and can be manipulated,
+     * but any attributes added through a subclass implementation won't be included.
+     *
+     * @return array
+     */
+    public function getAttributes()
+    {
+        $attributes = [];
+        if ($this->getIsImage()) {
+            $attributes = [
+                'width' => $this->getWidth(),
+                'height' => $this->getHeight(),
+                'alt' => $this->getTitle(),
+                'src' => $this->getURL()
+            ];
+
+            if ($this->IsLazyLoaded()) {
+                $attributes['loading'] = 'lazy';
+            }
+        }
+
+        $attributes = array_merge($attributes, $this->attributes);
+
+        if (isset($attributes['loading']) && $attributes['loading'] === 'eager') {
+            unset($attributes['loading']);
+        }
+
+        $this->extend('updateAttributes', $attributes);
+
+        return $attributes;
+    }
+
+    /**
+     * Custom attributes to process. Falls back to {@link getAttributes()}.
+     *
+     * If at least one argument is passed as a string, all arguments act as excludes, by name.
+     *
+     * @param array $attributes
+     * @return string
+     */
+    public function getAttributesHTML($attributes = null)
+    {
+        $exclude = null;
+
+        if (is_string($attributes)) {
+            $exclude = func_get_args();
+        }
+
+        if (!$attributes || is_string($attributes)) {
+            $attributes = $this->getAttributes();
+        }
+
+        $attributes = (array) $attributes;
+
+        $attributes = array_filter($attributes, function ($v) {
+            return ($v || $v === 0 || $v === '0');
+        });
+
+        if ($exclude) {
+            $attributes = array_diff_key(
+                $attributes,
+                array_flip($exclude)
+            );
+        }
+
+        // Create markup
+        $parts = [];
+
+        foreach ($attributes as $name => $value) {
+            if ($value === true) {
+                $value = $name;
+            } else {
+                if (is_scalar($value)) {
```

Comment:
I'd be included to just have `$value = (string) $value;` regardless of what $type value is

If you're passing in something like an object for whatever reason, it's probably a mistake and I see no reason to turn it into a json blob so that it displays "nicely".  If you really wanted to pass in json, you can just encode it before adding it as an attribute

Reply:
Same deal as thing as https://github.com/silverstripe/silverstripe-assets/pull/458#discussion_r665731990